### PR TITLE
Account id refactor

### PR DESCRIPTION
**Pull request by bachstatter**
This PR adds `id` to the data struct in `PerpsAccount`. This means we dont have to pass around `accountId` everywhere.
Given that we inherit the `createAccount` function from the core `AccountModule` we cant change the `createAccount` to set the id to the `PerpsAccount`. Instead we set the id (if it's missing) in the `modifyCollateral` method

### CODE REVIEW

**Review by fritzschoff — APPROVED** [review]

File: markets/perps-market/contracts/modules/PerpsAccountModule.sol [hunk 1]
```diff
@@ -44,6 +44,9 @@ contract PerpsAccountModule is IAccountModule {
         globalPerpsMarket.checkLiquidation(accountId);
 
         PerpsAccount.Data storage account = PerpsAccount.load(accountId);
+        if (account.id == 0) {
+            account.addAccountId(accountId);
```
Comment (sunnyvempati) [inline markets/perps-market/contracts/modules/PerpsAccountModule.sol:48]: can just do `account.id = accountId` and nix the `addAccountId` function

File: markets/perps-market/contracts/storage/PerpsAccount.sol [hunk 2]
```diff
@@ -103,6 +102,10 @@ library PerpsAccount {
         self.collateralAmounts[synthMarketId] += amountToAdd;
     }
 
+    function addAccountId(Data storage self, uint128 accountId) internal {
+        self.id = accountId;
+    }
```
Comment (sunnyvempati) [inline markets/perps-market/contracts/storage/PerpsAccount.sol:107]: x

**Review by sunnyvempati — APPROVED** [review]
one suggestion; pre approving

File: markets/perps-market/contracts/modules/PerpsAccountModule.sol (hunk 1 above)
Comment (bachstatter) [reply markets/perps-market/contracts/modules/PerpsAccountModule.sol:48]: I see, will do

**Pull request merged**